src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -139,23 +154,53 @@ class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig, protected
     //    This executable is our init-actions.sh, which will stand up our jupyter server and proxy.
     val initActions = Seq(new NodeInitializationAction().setExecutableFile(GcsPath(bucketName, GcsRelativePath(clusterResourcesConfig.initActionsScript)).toUri))
 
-    // Create a SoftwareConfig and set a property that makes the cluster have only one node
-    val softwareConfig = new SoftwareConfig().setProperties(Map("dataproc:dataproc.allow.zero.workers" -> "true").asJava)
+    val machineConfig = MachineConfig(clusterRequest.machineConfig, clusterDefaultsConfig)
+
+    // Create a config for the master node, if properties are not specified in request, use defaults
+    val masterConfig = new InstanceGroupConfig()
+      .setMachineTypeUri(machineConfig.masterMachineType.get)
```

Comment:
It's not possible to pass CPU and memory to the ClusterConfig object, it has to be passed in the form (example) "n1-standard-8". The swagger doc has a link to the machine type page: https://cloud.google.com/compute/docs/machine-types